core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt

```diff
@@ -621,11 +567,6 @@ class RawInfraImpl(
                     speedFromRoute
                 } else if (infraTagSpeed != null) {
```

Comment:
The whole `if` could now be changed to `val allowedSpeed = speedFromRoute ?: (infraTagSpeed ?: speedSection.default)`

It's not necessarily *better*, but it's an option

Reply:
I didn't like it with the default-speed logic, but now it looks better.
Amended the commit: https://github.com/OpenRailAssociation/osrd/compare/f60a9905e7491cdb9c92060bc0a94d65d81c472f..ae39abaf5205346613c6c1b7f58b801b33b480da